### DISCUSSION TITLE
.NET MAUI CollectionView with Staggered Layout

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#107] [Announcements]
# .NET MAUI CollectionView with Staggered Layout

Create a Staggered Layout for CollectionView in .NET MAUI.

https://vladislavantonyuk.github.io/articles/.NET-MAUI-CollectionView-with-Staggered-Layout/

**Comment by GeorgeLeithead** [comment]
Great article write up Vlad.  Interesting to see the use of the AndroidX controls within .net MAUI too.

**Comment by GeorgeLeithead** [comment]
Any thoughts/guides on how this could be used to automatically `span` multiple columns, depending on the width of the content and the available device width?  I'm thinking on Windows for example, you may want to have X columns as the device would support more... or if on Android, the device was in Landscape mode and could then support 4-X columns?

**Reply by VladislavAntonyuk** [reply]
Under the hood it uses default Android StaggeredGridLayoutManager. You can change Span number using .NET MAUI OnPlatform or Visual States.

The article currently is a start point. I didn’t test code on iOS and for Windows it needs to adapt CommunityToolkit StaggeredLayout to .NET MAUI ListViewBase
